dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,148 @@
+[[getting-traffic-into-cluster]]
+= Getting Traffic Into The Cluster
+{product-author}
+{product-version]
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
+== Overview
+There are many ways to access the cluster. This section describes some
+commonly used approaches.
+
+The recommendation is:
+
+- If you have HTTP/HTTPS -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have Encrypted TCP (i.e., TLS with the SNI header) -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have TCP or UDP -- Use xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-externalIP[ExternalIP], or
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-nodeport[NodePort].
+
+The router is the most common way to access the cluster.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+TCP or UDP is useful when the HTTP protocol is not being used or non-standard
+ports are in use. There is more manual setup and monitoring involved.
+
+The admin is responsible for configuring the external network so that
+the DNS name lands on the exposed IP. When setting up a router it can be
+convenient to use wildcard DNS since this lets the users self-provision
+the router.
```

Comment:
On 09/13/2016 03:56 PM, Dan Winship wrote:
>
> In dev_guide/getting_traffic_into_cluster.adoc 
> <https://github.com/openshift/openshift-docs/pull/2827#discussion_r78632162>:
>
> > +- If you have HTTP/HTTPS -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
> > +- If you have Encrypted TCP (i.e., TLS with the SNI header) -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
> > +- If you have TCP or UDP -- Use xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
> > +xref:../dev_guide/getting_traffic_into_cluster.adoc#using-externalIP[ExternalIP], or
> > +xref:../dev_guide/getting_traffic_into_cluster.adoc#using-nodeport[NodePort].
> > +
> > +The router is the most common way to access the cluster.
> > +This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
> > +
> > +TCP or UDP is useful when the HTTP protocol is not being used or non-standard
> > +ports are in use. There is more manual setup and monitoring involved.
> > +
> > +The admin is responsible for configuring the external network so that
> > +the DNS name lands on the exposed IP. When setting up a router it can be
> > +convenient to use wildcard DNS since this lets the users self-provision
> > +the router.
>
> It's not at all clear what that means if you don't already know 
> everything about routers.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openshift/openshift-docs/pull/2827/files/fb88bc0f87d4b52a7b951a4f82b17a60f6b9e2b1#r78632162>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ANUgenK63k_otJ7RuebJuQ6fsqk2kcKlks5qpv_TgaJpZM4J6qBl>.
>
Hi Dan,
I reworded this in the next version (I will let you know when it is pushed).
phil